### PR TITLE
CLN: Removed unreached else block GH33478

### LINKED ISSUE #33478: CLN remove unreachable code in pandas/core/groupby/generic.py::DataFrameGroupBy::_wrap_applied_output
NOTE: I'd originally [opened a PR](https://github.com/pandas-dev/pandas/pull/32583) to address this, but am now busy with other obligations + other PRs to respond to. So I'm opening it up as a good first issue for now, and will return to work on it if nobody takes it.

---------

In `pandas/core/groupby/generic.py::DataFrameGroupBy::_wrap_applied_output`, it seems this branch is never reached:

```
        else:
            # Handle cases like BinGrouper
            return self._concat_objects(keys, values, not_indexed_same=not_indexed_same)
```

Two things need to be done here:

1. refactor `_wrap_applied_output` by making a helper function out of some of its internals (it's currently very long, and changes are hard to review)
2. Remove this unused `else` branch

### PR DESCRIPTION
**Pull request by FollowTheProcess**
Unreached else block in method _wrap_applied_output was removed.

Local tests passed before and after change

- [ ] xref #33478 (Not fully closed, this addresses 1 part)
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1373,10 +1373,6 @@ def _wrap_applied_output(self, keys, values, not_indexed_same=False):
                 # of columns
                 return self.obj._constructor_sliced(values, index=key_index)
 
-        else:
```
Comment (MarcoGorelli) [inline pandas/core/groupby/generic.py:1376]: Thanks @FollowTheProcess - while you're here, can you change the `elif` in line 1221 to `else`?

**Comment by MarcoGorelli** [comment]
> closes #33478 (Not fully closed, this addresses 1 part)

Can you change "closes" to "xref"?

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (FollowTheProcess) [reply pandas/core/groupby/generic.py:1376]: Sorry yep, will do this and the xref now

Comment (FollowTheProcess) [reply pandas/core/groupby/generic.py:1376]: That's now done :) Thanks for helping me contribute (in an albeit microscopic way) the contributing docs made it super easy as a newbie. I hope to be able to contribute more as I learn more.

**Comment by MarcoGorelli** [comment]
The `Windows py37_np18` shows a failure (`pandas/tests/base/test_unique.py::test_unique_bad_unicode[idx_or_series_w_bad_unicode0]`), although if I check out your pull request and run it myself it passes (also, none of the functions in `pandas/tests/base/test_unique.py` hit this code path), so am pretty sure the CI failure is unrelated

**Comment by FollowTheProcess** [comment]
Oh, weird! I ran the full test suite (test_fast.sh) before committing and no failures for me either. Let me know if you need me to do anything else

**Comment by jreback** [comment]
thanks @FollowTheProcess

**Comment by FollowTheProcess** [comment]
Thanks both for the help. I hope to contribute more in the future!

**Pull request merged**